madrid/bicimad.py

```diff
@@ -0,0 +1,147 @@
+import requests
+import pandas as pd
+
+from madrid.auth_credentials import auth_bicimad
+
+
+def shout(func):
+    def wrapper(*args, **kwargs):
+        print(f'Executing {func.__name__}')
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+class BiciMad:
+    """
+    Class in charge of connecting to BiciMad API and querying information
+    Documentation here: https://apidocs.emtmadrid.es/
+    """
+    url_login = 'https://openapi.emtmadrid.es/v1/mobilitylabs/user/login/'
+    url_logout = 'https://openapi.emtmadrid.es/v1/mobilitylabs/user/logout/'
+    url_who_am_i = 'https://openapi.emtmadrid.es/v1/mobilitylabs/user/whoami/'
+    url_get_all_stations_info = 'https://openapi.emtmadrid.es/v1/transport/bicimad/stations/'
+
+    stations_keys = ['id', 'number', 'activate', 'total_bases', 'dock_bikes', 'free_bases', 'reservations_count']
+
+    def __init__(self):
+        self._load_access_token()
+
+    @staticmethod
```

Comment:
I tried reading about it, but can you tell me what `staticmethod` does?

Reply:
It is a method that does not use `self`, so in principle could be defined out of the class, as an independent function. But it is left inside for keeping the code toguether.